File: src/core/extractors/actors.ts
import * as vscode from 'vscode'
import { callLLMJson } from '../../llm/client'
import { Actor } from '../models/outputs'

const SYSTEM = `You are a business analyst extracting actors from a BRD.
An actor is any person, system, or role that interacts with the system.
Return JSON: { "actors": [{ "name": "...", "description": "..." }] }`

export async function extractActors(
  brdText: string,
  token?: vscode.CancellationToken
): Promise<Omit<Actor, 'id'>[]> {
  const result = await callLLMJson<{ actors: Omit<Actor, 'id'>[] }>(
    brdText,
    SYSTEM,
    token
  )
  return result.actors
}

File: pythonlibrary/aibrd/extractors/actors.py
from ..llm.client import call_llm_json

_SYSTEM = """You are a business analyst extracting actors from a BRD.
An actor is any person, system, or role that interacts with the system.
Return JSON: {"actors": [{"name": "...", "description": "..."}]}"""


def extract_actors(brd_text: str) -> list[dict]:
    result = call_llm_json(brd_text, _SYSTEM)
    return result.get("actors", [])
